theano/tensor/nnet/abstract_conv.py

```diff
@@ -1517,18 +1525,23 @@ def conv(self, img, kern, mode="valid", dilation=1):
                      ] = kern
 
         if self.convdim == 2:
+            if img.shape[1] % self.num_groups != 0:
+                raise ValueError("Number of input channels must be divisible by num_groups")
             val = _valfrommode(mode)
             bval = _bvalfromboundary('fill')
+            input_channel_offset = img.shape[1] // self.num_groups
+            output_channel_offset = kern.shape[0]
```

Comment:
For grouped convolution, the input would need to be divided into `num_groups` different groups. So for each group the input would be of shape (batches, in_channels // num_groups, ih, iw) .

 The kernel shape for each group would be (num_filters // num_groups, in_channels // num_groups , kh, kw). Now, for the entire kernel, the individual  kernels from each group would need to be combined either in the first or the second dimension. Concatenating using the second dimension would result in a kernel of shape (numfilters // num_groups , in_channels, kh, kw) which is done in the first method. 

Combining using the first dimension would result in a kernel of shape (num_filters, in_channels // num_groups , kh, kw) which is the second method.